tests/cacl/test_cacl_application.py

```diff
@@ -613,22 +811,67 @@ def test_multiasic_cacl_application(duthosts, rand_one_dut_hostname, localhost,
+
+    actual_iptables_rules = duthost.command("iptables -S")["stdout_lines"]
+
+    # Ensure all expected iptables rules are present on the DuT
+    logger.info("Number of expected iptable rules:{}, number of acutal iptables rules:{}, number of ignored_iptable_rules_v4 rules:{}"
+                .format(len(set(expected_iptables_rules)), len(set(actual_iptables_rules)), len(set(ignored_iptable_rules_v4))))
+
+    missing_iptables_rules = set(expected_iptables_rules) - set(actual_iptables_rules)
+    pytest_assert(len(missing_iptables_rules) == 0, "Missing expected iptables rules: {}".format(repr(missing_iptables_rules)))
+
+    # Ensure there are no unexpected iptables rules present on the DuT
+    unexpected_iptables_rules = set(actual_iptables_rules) - set(expected_iptables_rules) - set(ignored_iptable_rules_v4)
+    pytest_assert(len(unexpected_iptables_rules) == 0, "Unexpected iptables rules: {}".format(repr(unexpected_iptables_rules)))
+
+def test_cacl_scale_rules_ipv6(duthosts, rand_one_dut_hostname, collect_ignored_rules, clean_scale_rules):
+    """
+    Test case to ensure cover scale rules for control plan ACL for ipv6
+
+    This is done by collecting existing ip6table rules as ingnored rules list, creating scale rules for SNMP-ACL, SSH-ONLY, NTP-ACL tables
+    and generating our own set of expected ip6tables rules based on the DUT's configuration and comparing them against the actual ip6tables
+    rules on the DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    ignored_iptable_rules_v6 = collect_ignored_rules["v6"]
+
     generate_scale_rules(duthost, "ipv6")
-    verify_cacl(duthost, localhost, creds, docker_network)
-    clean_scale_rules(duthost)
\ No newline at end of file
+
+    expected_ip6tables_rules = generate_expected_cacl_rules(duthost, "ipv6")
+
+    # Append rules which block "ip2me" traffic on p2p interfaces, only for null multi-asic
+    generate_and_append_block_ip2me_traffic_rules(duthost, [], expected_ip6tables_rules, None)
+
+    actual_ip6tables_rules = duthost.command("ip6tables -S")["stdout_lines"]
+
+    # Ensure all expected ip6tables rules are present on the DuT
+    missing_ip6tables_rules = set(expected_ip6tables_rules) - set(actual_ip6tables_rules)
+    pytest_assert(len(missing_ip6tables_rules) == 0, "Missing expected ip6tables rules: {}".format(repr(missing_ip6tables_rules)))
+
+    # Ensure there are no unexpected ip6tables rules present on the DuT
+    logger.info("Number of expected ip6table rules:{}, number of acutal ip6tables rules:{}, number of ignored_iptable_rules_v6 rules:{}"
+                .format(len(set(expected_ip6tables_rules)), len(set(actual_ip6tables_rules)), len(set(ignored_iptable_rules_v6))))
+
+    unexpected_ip6tables_rules = set(actual_ip6tables_rules) - set(expected_ip6tables_rules) - set(ignored_iptable_rules_v6)
+    pytest_assert(len(unexpected_ip6tables_rules) == 0, "Unexpected ip6tables rules: {}".format(repr(unexpected_ip6tables_rules)))
```

Comment:
Please add an extra empty line at the end of file.

Reply:
Done